app/models/user.rb

```diff
@@ -29,4 +29,18 @@ def registered_user?
     roles.exists?(name: "registered_user")
   end
 
+  def blocked_user?
+    roles.exists?(name: "blocked_user")
+  end
+
+  def deactivate
+    user_roles.destroy_all
+    user_roles.create(role: Role.find_by(name: "blocked_user"))
```

Comment:
I like how you're using find_by to ensure that you're not creating a new user_role here that will by pass permissions. Spiffy.